Wikipedia/Code/InsertMediaSearchResultsCollectionViewController.swift

```diff
@@ -0,0 +1,230 @@
+import UIKit
+import SafariServices
+
+fileprivate class FlowLayout: UICollectionViewFlowLayout {
+    private var oldBoundsWidth: CGFloat = 0
+
+    override init() {
+        super.init()
+        minimumInteritemSpacing = 8
+        minimumLineSpacing = 32
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+        defer {
+            oldBoundsWidth = newBounds.width
+        }
+        return super.shouldInvalidateLayout(forBoundsChange: newBounds) || newBounds.width != oldBoundsWidth
+    }
+
+    override func invalidateLayout(with context: UICollectionViewLayoutInvalidationContext) {
+        defer {
+            super.invalidateLayout(with: context)
+        }
+
+        guard let collectionView = collectionView else {
+            return
+        }
+
+        let dimension = (collectionView.bounds.width / 3) - minimumInteritemSpacing * 2
```

Comment:
When rotating, the cells overlap the safe area because it's not taken into account here - the flow layout's `sectionInset` left & right should be updated to be the `scrollIndicatorInsets` left and right and those values should be subtracted from the width before calculating dimension